src/wayland-eglsurface.c

```diff
@@ -1126,33 +1462,86 @@ pop_acquired_image(WlEglSurface *surface)
 }
 
 static EGLint
-add_surface_image(WlEglDisplay *display, WlEglSurface *surface)
+init_surface_image(WlEglDisplay *display, WlEglSurface *surface,
+                   WlEglStreamImage    *image)
 {
     WlEglPlatformData   *data     = display->data;
     EGLDisplay           dpy      = display->devDpy->eglDisplay;
+
+    image->eglImage = data->egl.createImage(dpy, EGL_NO_CONTEXT,
+                                            EGL_STREAM_CONSUMER_IMAGE_NV,
+                                            (EGLClientBuffer)surface->ctx.eglStream,
+                                            NULL);
+    if (image->eglImage == EGL_NO_IMAGE_KHR) {
+        return EGL_BAD_ALLOC;
+    }
+
+    /*
+     * Create a per-stream image release timeline.
+     *
+     * This is needed since we will be the ones signaling acquire points. If the acquire points
+     * are on the same timeline as the release points then they will accidentally signal all
+     * pending release points.
+     */
+    if (surface->wlSyncobjSurf) {
+        struct wp_linux_drm_syncobj_manager_v1 *wrapper;
+
+        /* Create a DRM timeline and share it with the compositor */
+        if (drmSyncobjCreate(display->drmFd, 0, &image->drmSyncobjHandle)) {
+            goto fail;
+        }
+
+        if (drmSyncobjHandleToFD(display->drmFd, image->drmSyncobjHandle,
+                                 &image->drmSyncobjFd)) {
+            goto fail;
+        }
+        image->acquireSync = EGL_NO_SYNC_KHR;
+
+        /* Get a DRM timeline wl object */
+        wrapper = wl_proxy_create_wrapper(display->wlDrmSyncobj);
+        wl_proxy_set_queue((struct wl_proxy *)wrapper, surface->wlEventQueue);
```

Comment:
Why do we need to do this?

Reply:
This was leftover from that bug when I ran into firefox committing from other threads. Removed.